specifyweb/frontend/js_src/lib/components/PickLists/index.tsx

```diff
@@ -30,166 +27,180 @@ export function PickListComboBox({
-      console.error('Duplicate picklist entries found', { items, resource });
-  }, [items, resource]);
+    const itemValues = normalizedItems.map(({ value }) => value) ?? [];
+    if (itemValues.length !== new Set(itemValues).size) {
+      console.error('Duplicate picklist entries found', {
+        normalizedItems,
+        resource,
+      });
+    }
+  }, [normalizedItems, resource]);
 
   const [pendingNewValue, setPendingNewValue] = React.useState<
     string | undefined
   >(undefined);
 
+  // If a pending new value appears in items (after it’s added), write it to the model
   React.useEffect(
     () =>
       typeof pendingNewValue === 'string' &&
-      items.some(({ value }) => value === pendingNewValue)
-        ? updateValue(pendingNewValue)
+      normalizedItems.some(({ value }) => value === pendingNewValue)
+        ? writeModelValue(pendingNewValue)
         : undefined,
-    [items, pendingNewValue, updateValue]
+    [normalizedItems, pendingNewValue, writeModelValue]
   );
 
-  function addNewValue(value: string): void {
-    if (pickList?.get('type') === PickListTypes.FIELDS) updateValue(value);
-    else if (pickList?.get('type') === PickListTypes.ITEMS)
-      setPendingNewValue(value);
-    else throw new Error('Adding item to wrong type of picklist');
+  function handleRequestAddNewValue(typedValue: string): void {
+    if (pickList?.get('type') === PickListTypes.FIELDS) {
+      writeModelValue(typedValue);
+    } else if (pickList?.get('type') === PickListTypes.ITEMS) {
+      setPendingNewValue(typedValue);
+    } else {
+      throw new Error('Adding item to wrong type of picklist');
+    }
   }
 
-  const currentValue = items.find((item) => item.value === value);
-  const isExistingValue = typeof currentValue === 'object';
+  const matchedCurrentItem = normalizedItems.find(
+    (item) => item.value === currentValueAsString
+  );
+  const isCurrentValueInList = typeof matchedCurrentItem === 'object';
 
-  const autocompleteItems = React.useMemo(
+  const autoCompleteSourceItems = React.useMemo(
     () =>
-      items
+      normalizedItems
         .filter(({ value }) => Boolean(value))
         .map((item) => ({
           label: item.title,
           data: item.value,
         })),
-    [items]
+    [normalizedItems]
   );
 
-  const handleAdd = hasToolPermission('pickLists', 'create')
-    ? rawHandleAdd
+  const canAddToPicklist = hasToolPermission('pickLists', 'create')
+    ? handleAddFromCaller
     : undefined;
 
-  const name = pickList?.get('name') ?? pickListName;
-
+  const fieldNameForForm = pickList?.get('name') ?? pickListName;
   const isReadOnly = React.useContext(ReadOnlyContext);
+
+  const isRankOrTreeLevelPickList = React.useMemo(() => {
+    const identifierText = [pickList?.get?.('name'), pickListName, (field as any)?.name]
+      .filter(Boolean)
+      .join(' ')
+      .toLowerCase();
+    return /(rank|taxon.*level|geo.*level|_treelevelcombobox)/i.test(identifierText);
```

Comment:
Why not just render all special pick lists as read-only?

https://github.com/specify/specify7/pull/7400/files#diff-6bae6d2d398090d7c917d80ff60ef98c9c15e765669841b5a2b150069acf8e53R166

See
https://github.com/specify/specify7/blob/b45cef28f3c8a5109ee132f8c26a18b78d020f52/specifyweb/frontend/js_src/lib/components/FormFields/ComboBox.tsx#L40-L45

Reply:
See this PR: https://github.com/specify/specify7/pull/7400